odoo/addons/base/tests/test_menu.py

```diff
@@ -28,3 +28,60 @@ def test_00_menu_deletion(self):
 
         orphans =  Menu.search([('id', 'in', all_ids), ('parent_id', '=', False)], order="id")
         self.assertEqual([child1.id, child2.id], orphans.ids)
+
+    def test_best_root_menu_for_model(self):
+        Menu = self.env['ir.ui.menu']
+        Action = self.env['ir.actions.act_window']
+        menu_cpt = 0
+
+        def new_menu(name, parent_id=None, action=None):
+            nonlocal menu_cpt
+            menu_cpt += 1
+            inst = Menu.create({'name': name, 'parent_id': parent_id})
+            if action:
+                inst.action = action
+            return inst
+
+        def new_action(name, res_model, view_mode=None, domain=None, context=None):
+            return Action.create(dict((k, v) for k, v in
+                                      dict(name=name, res_model=res_model, view_mode=view_mode, domain=domain,
+                                           context=context, type='ir.actions.act_window').items() if v))
```

Comment:
Do you really need this logic ? I think if value are None the create method will just ignore them
```python
Action.create({
    'name': name,
    'res_model': res_model,
    'view_mode': view_mode,
    'domain': domain,
    'context': context,
    'type': 'ir.actions.act_window'
})
```